doc/conf.py

```diff
@@ -551,6 +551,22 @@ def setup(app):
     },
 }
 
+needs_global_options = {
```

Comment:
What is all this?

Reply:
Ok I cleaned this up. I'll resolve this comment and re-request a review since I've pushed so many changes up since your first review!